site-src/geps/gep-1867.md

```diff
@@ -0,0 +1,128 @@
+
+Today `GatewayClass.spec.parametersRef` is available to attach arbitrary configuration to a `GatewayClass`.
+
+This GEP will explain why that is not sufficient to meet common use cases, and introduce a new field - `infrastructure` - to address these cases.
+
+Related discussions:
+* [Support cluster-local Gateways](https://github.com/kubernetes-sigs/gateway-api/discussions/1247)
+* [Scaling Gateway Resources](https://github.com/kubernetes-sigs/gateway-api/discussions/1355)
+* [Manual deployments](https://github.com/kubernetes-sigs/gateway-api/issues/1687)
+* [Merging Gateways](https://github.com/kubernetes-sigs/gateway-api/pull/1863)
+* [In Cluster Gateway Deployments])(https://github.com/kubernetes-sigs/gateway-api/pull/1757)
+
+## Goals
+
+* Provide the ability to configure arbitrary (implementation specific) attributes about a **specific Gateway**.
+* Provide the ability to configure a standardized set of attributes about a **specific Gateway**.
+
+## Challenges with GatewayClass
+
+`GatewayClass.spec.parametersRef` is the existing mechanism to configure arbitrary fields on a Gateway.
+However, this introduces operational challenges when configuring Gateways.
+
+### Scope
+
+As a `Gateway` manager (with RBAC permissions to a specific `Gateway`) I should be able to declaratively make changes to that `Gateway` without the need for access to cluster-scoped resources (`GatewayClass`) and without affecting other `Gateways` managed by the same `GatewayClass`.
+
+As a cluster scoped resource, `GatewayClass` does not meet this requirement.
+This restricts customization use cases to either a few pre-provisioned classes by the admin, or running in an environment where the "Infrastructure Provider" and "Cluster Operator" are the same roles.
+The distinction between these roles is explicitly called out on the homepage: https://gateway-api.sigs.k8s.io/#what-is-the-gateway-api.
+
+### Custom Resource
+
+`parametersRef` is entirely a generic implementation-specific meaning.
+This means users will either need a customer CRD or use untyped resources like ConfigMap.
+Neither of these have any consistency between implementations.
+While there will always be some vendor-specific requirements, there are also a number of configuration aspects of a Gateway that are common between implementations.
+However, these cannot currently be expressed in a vendor-neutral way.
+
+Additionally, there is hesitancy to use a CRD (which leads to CRD proliferation), which pushes users towards untyped ConfigMaps which are not much better than annotations. The scoping, as mentioned above, is also a bit awkward of a cluster scoped resource pointing to a namespaced object.
+
+### Separation of concerns
+
+While there is value out of providing class-wide options as defaults, there is also value in providing these options on the object (Gateway) directly.
```

Comment:
This is discussed in the API spec itself, see L92

Reply:
I think this is a very important question because this GEP introduces a lot of overlap between the two resources. @howardjohn do you have a link to the API spec you're referring to? The closest I could find was the GatewayClass spec, but it doesn't really seem to cover merging.

https://github.com/kubernetes-sigs/gateway-api/blob/7b8f168c82a90522209a340bf291fcacab0fc5ef/apis/v1beta1/gatewayclass_types.go#L82-L94